README.md

```diff
@@ -35,18 +37,67 @@ Once the file is created, be sure to allow read access to `addresses.json` insid
 fs_permissions = [{ access = "read", path = "./addresses/addresses.json"}]
 ```
 
-### Step 4: Create a Proposal
+#### Step 4: Create a Proposal
 
 Choose a model that fits your needs:
 
 -   [Multisig Proposal](docs/guides/multisig-proposal.md)
 -   [Timelock Proposal](docs/guides/timelock-proposal.md)
 -   [Governor Bravo Proposal](docs/guides/governor-bravo-proposal.md)
 
-### Step 5: Implement Scripts and Tests
+#### Step 5: Implement Scripts and Tests
 
 Create scripts and/or tests. Check [Guides](docs/guides/multisig-proposal.md) and [Integration Tests](docs/testing/integration-tests.md).
 
+### Type Checking
+
+Type checking allows verification of deployed bytecode on any contracts with the bytecode present in local artifacts. With this feature, developer `A` can easily deploy some contracts, and developer `B` can verify `A`'s deployments by simply running the type checking script. Additionaly, `A` can also use this feature to verify their own deployments. `A` can take the following steps:
+
+-   Follow the steps 1 to 3 on [Proposal Simulation](#proposal-simulation) section
+-   Add the deployed contracts to `Addresses.json`.
+-   Create a `TypeCheckAddresses.json` file following the instructions provided in [type-check.md](docs/guides/type-check.md).
+-   Enter `lib/forge-proposal-simulator/typescript` directory and install npm packages.
+
+```bash
+cd lib/forge-proposal-simulator/typescript && npm i
+```
+
+-   Change directory again to the root repo
+
+```bash
+cd ../../../
+```
+
+-   Add below environment variables to `.env`.
+
+```
+ADDRESSES_PATH                # Path to addresses.json file
+TYPE_CHECK_ADDRESSES_PATH     # Path to typeCheckAddresses.json file
+ARTIFACT_PATH                 # Path of artifact folder
+```
+
+Example:
+
+```
+TYPE_CHECK_ADDRESSES_PATH=addresses/TypeCheckAddresses.json
+ADDRESSES_PATH=addresses/Addresses.json
+ARTIFACT_PATH=out/
+```
+
+-   Make sure to allow read access to `Addresses.json`, `TypeCheckAddresses.json` and `artifact` folder inside of `foundry.toml`.
+
+```toml
+[profile.default]
+
+fs_permissions = [{ access = "read", path = "./"}]
+```
+
+-   Run the following command on root repo to type check all contracts added in `TypeCheckAddresses.json`.
```

Comment:
These are the steps to run from root so that's why root is correct here. I have added another section to run example contracts within FPS here [ddea628](https://github.com/solidity-labs-io/forge-proposal-simulator/pull/47/commits/ddea628b95ec7a37d1e45eb3c7de610107228e88)